3rdparty/communi/src/core/ircmessage.cpp

```diff
@@ -701,7 +701,7 @@ IrcMessage* IrcMessage::fromData(const QByteArray& data, IrcConnection* connecti
     if (!tag.isEmpty()) {
         QDateTime ts = QDateTime::fromString(QString::fromUtf8(tag), Qt::ISODate);
         if (ts.isValid())
-            message->d_ptr->timeStamp = ts.toTimeSpec(Qt::LocalTime);
+            message->d_ptr->timeStamp = ts.toLocalTime();
```

Comment:
One issue with this change is that it is code that is copied verbatim from https://github.com/communi/libcommuni (with the exception of the `/3rdparty/communi/communi.pri`file) - but that hasn't been updated for over 4 years - unlike a couple of forks which have seen changes most recently 4 months ago - it might be work reviewing them to see what improvements they've made and replacing the code with that from one of the forks if they have this fix in place as well):
* https://github.com/Chatterino/libcommuni
* https://github.com/Nerixyz/libcommuni

Reply:
> Are you saying the old code was copied from somewhere else and changing it is a risk that could conflict with the dependency in place?

It ***is*** copied from somewhere else but it is not formally a dependency as it is a copy - however if upstream were or has been updated we wouldn't get that update unless we recopied it ourselves (it isn't handled automatically). As it happens at least one fork has made that change (see https://github.com/Chatterino/libcommuni/tree/chatterino-cmake ) but JPNurmi (who is the original author) does seem to have done anything since 2021)
<img width="1366" height="730" alt="image" src="https://github.com/user-attachments/assets/896d8f61-c632-40b6-a247-bf38ab354482" />